pkg/fqdn/bootstrap/fqdn_bootstraper.go

```diff
@@ -122,8 +98,7 @@ func (b *fqdnProxyBootstrapper) BootstrapFQDN(possibleEndpoints map[uint16]*endp
 		ConcurrencyGracePeriod: option.Config.DNSProxyConcurrencyProcessingGracePeriod,
 	}
 	var dnsProxy fqdnproxy.DNSProxier
-	dnsProxy, err = dnsproxy.StartDNSProxy(dnsProxyConfig, b.lookupEPByIP, b.ipcache.LookupSecIDByIP, b.ipcache.LookupByIdentity,
-		b.notifyOnDNSMsg)
+	dnsProxy, err = dnsproxy.StartDNSProxy(dnsProxyConfig, b.lookupEPByIP, b.ipcache.LookupSecIDByIP, b.ipcache.LookupByIdentity, b.dnsRequestHandler.NotifyOnDNSMsg)
```

Comment:
This is neat already, but would it be even nicer if we could inject the handler as a dependency of the proxy instead of via these function pointers?

Reply:
Fully agree. But I'd like to keep the changes of this PR as small as possible. Probably worth to further refactor the internals in follow up PRs if this is ok for you

Reply:
yep, out of scope for this PR 👍